source/extensions/filters/udp/dns_filter/dns_parser.cc

```diff
@@ -66,7 +66,20 @@ void DnsQueryRecord::serialize(Buffer::OwnedImpl& output) {
   output.writeBEInt<uint16_t>(class_);
 }
 
-bool DnsMessageParser::parseDnsObject(const Buffer::InstancePtr& buffer) {
+DnsQueryContextPtr DnsMessageParser::createQueryContext(Network::UdpRecvData& client_request) {
+  DnsQueryContextPtr query_context = std::make_unique<DnsQueryContext>(
+      client_request.addresses_.local_, client_request.addresses_.peer_);
+
+  query_context->status_ = parseDnsObject(query_context, client_request.buffer_);
+  if (!query_context->status_) {
+    ENVOY_LOG(error, "Unable to parse buffer into a DNS object");
```

Comment:
would we want to add more context in this log specifying the client_request addresses?

Reply:
Yeah, I can add that.